wire.go

```diff
@@ -0,0 +1,68 @@
+package opentracing
+
+// WireEncoding identifies a WireInjector/WireExtractor pair.
+type WireEncoding int
+
+const (
+	// WIRE_ENCODING_SPLIT_BINARY divides the encoded Span into two components:
+	//
+	//   1) The "tracer state" for the Span (for example, in Dapper this would
+	//      include a trace_id, a span_id, and a bitmask representing the
+	//      sampling status for the given trace)
+	//   2) Any Trace Attributes (per Span.SetTraceAttribute)
+	//
+	// The encoded data is separated in this way for a variety of reasons; the
+	// most important is to give OpenTracing users a portable way to opt out of
+	// Trace Attribute propagation entirely if they deem it a stability risk.
+	//
+	// The `carrier` for injection must be two pointers to `[]byte`s for the
```

Comment:
that would be my preference, as it makes the API for the end user compiler-enforced, and still keeps the mechanism extensible (I didn't even notice that you had `...` there). For common cases like two-maps, one map, binary, etc. we can provide standard factory methods in the utils.

One drawback of the interface approach is that it requires new object allocation, but it can be address via object pool.